src/fromager/commands/build.py

```diff
@@ -170,8 +172,19 @@ def _is_wheel_built(
 
     try:
         logger.info(f"{req.name}: checking if {req} was already built")
-        sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
-        return True
+        url, _ = sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
+        build_tag = wkctx.settings.build_tag(req.name, dist_version)
+        wheel_filename = urlparse(url).path.rpartition("/")[-1]
+        _, _, build_tag_from_name, _ = parse_wheel_filename(wheel_filename)
+        if build_tag_from_name:
+            if build_tag_from_name[0] > build_tag:
+                raise ValueError(
+                    f"{dist_name}: changelog for version {dist_version} is inconsistent. Found build tag {build_tag_from_name[0]} but expected {build_tag}"
+                )
+            return build_tag_from_name[0] == build_tag
+
+        # build tag == 0 is equivalent to having no build tag
+        return build_tag == 0
```

Comment:
This logic is correct, but not as clear as it could be. See what you think of these changes.

```suggestion
        _, _, build_tag_from_name, _ = parse_wheel_filename(wheel_filename)
        if build_tag_from_name:
          existing_build_tag = build_tag_from_name[0]
        else:
          existing_build_tag = 0  # default to 0, even if it is not in the name

        if existing_build_tag > build_tag:
            raise ValueError(
                    f"{dist_name}: changelog for version {dist_version} is inconsistent. Found build tag {existing_build_tag} but expected {build_tag}"
                )

        return existing_build_tag == build_tag
```

Reply:
I like it, will update